sympy/core/expr.py

```diff
@@ -990,6 +990,15 @@ def _eval_endpoint(left):
 
         value = B - A
 
+        # Catch undefined forms like oo - oo or oo * symbolic
+        if isinstance(value, Add) and any(arg.is_infinite for arg in value.args):
+            if any(a.is_infinite for a in value.args) and any(a.has(x) for a in value.args):
+                return S.NaN
+
+        if isinstance(value, Mul) and any(arg.is_infinite for arg in value.args):
+            if any(arg.has(x) for arg in value.args if arg != S.Infinity and arg != S.NegativeInfinity):
+                return S.NaN
```

Comment:
I don't understand why you have added this. Why not just remove it?

It doesn't really fix anything in a robust way.

Reply:
Okay.